core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -72,6 +92,66 @@
+         builder.allowBlockingCallsInside(SecureRandom.class.getName(), "nextBytes");
+         // UDP calls PlainDatagramSocketImpl#send which can block if OS buffer is full - but we just ignore
+         builder.allowBlockingCallsInside(UDP.class.getName(), "_send");
+         builder.allowBlockingCallsInside(FlowControl.class.getName(), "adjustCredit");
+         // Random JGroups stuff
+         builder.allowBlockingCallsInside(Table.class.getName(), "add");
+         // Just assume all the thread pools don't block - not we check the rejected handler below though
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "getTask");
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "processWorkerExit");
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "execute");
+         builder.allowBlockingCallsInside(ScheduledThreadPoolExecutor.class.getName(), "schedule");
+
+         // TODO: This can actually block us for a bit, we should really remove this!!!
+         builder.allowBlockingCallsInside(StateTransferLockImpl.class.getName(), "acquireSharedTopologyLock");
+         builder.allowBlockingCallsInside(StateTransferLockImpl.class.getName(), "releaseSharedTopologyLock");
+
+         // LimitedExecutor just submits a task to another thread pool
+         builder.allowBlockingCallsInside(LimitedExecutor.class.getName(), "execute");
+
+         // Some test utilities may block the cpu thread - which is fine
+         builder.allowBlockingCallsInside(EvictionWithConcurrentOperationsTest.class.getName() + "$Latch", "blockIfNeeded");
+         builder.allowBlockingCallsInside(CheckPoint.class.getName(), "await");
+         builder.allowBlockingCallsInside(CheckPoint.class.getName(), "trigger");
+         builder.allowBlockingCallsInside(BlockingInterceptor.class.getName(), "blockIfNeeded");
+         builder.allowBlockingCallsInside(TestingUtil.class.getName(), "sleepRandom");
+         builder.allowBlockingCallsInside(TestingUtil.class.getName(), "sleepThread");
+
+         // The blocking iterator locks to signal at the end - ignore
+         builder.allowBlockingCallsInside(BlockingFlowableIterable.class.getName() + "$BlockingFlowableIterator", "signalConsumer");
+
+         // Allow logging to block
+         builder.allowBlockingCallsInside(Logger.class.getName(), "logf");
+         builder.allowBlockingCallsInside(java.util.logging.Logger.class.getName(), "log");
+
+         // Make sure our rejection handlers never block in a non blocking thread
+         builder.disallowBlockingCallsInside(BlockingRejectedExecutionHandler.class.getName(), "rejectedExecution");
+         builder.disallowBlockingCallsInside(NonBlockingRejectedExecutionHandler.class.getName(), "rejectedExecution");
+
+         builder.blockingMethodCallback(bm -> {
+            throw new CacheException(String.format("Blocking call! %s on thread %s", bm, Thread.currentThread()));
```

Comment:
Maybe we should remove this or convert it to a different Assertion or something.

Reply:
Will leave this for now.

Reply:
AssertionError would be good, but I'd also report it as a failure directly in case the test expects an exception and doesn't bother to check the exception type.

Something like: https://github.com/infinispan/infinispan/blob/5bf76758e34cf487ff0423ec72a9d9f58a76a61b/commons-test/src/main/java/org/infinispan/commons/test/RunningTestsRegistry.java#L72

Reply:
Sure, let me look into this. It would be good because then I can explicitly say the method and test, where as currently it just shows in some tests as a timeout :(